Macros/Macros.js

```diff
@@ -107,6 +215,56 @@ class FurnaceMacros {
         }
     }
 
+    // request execution of macro as a GM
+    async executeMacroAsGM(macro, context) {
+        const activeGMs = game.users.entities.filter(u => u.isGM && u.active);
+        if (activeGMs.length === 0) {
+            ui.notifications.error(`There are no connected GMs to run the macro ${macro.name} in the GM context.`);
+            return "";
+        }
+        // Elect a GM to run the Macro
+        const electionResponse = await new Promise((resolve, reject) => {
+            const requestId = randomID();
```

Comment:
This really does not guarantee uniqueness - I know it is really likely to. Since it is cheap I would suggest
userID + time + randomID

Reply:
The possibility of a `Math.random()` giving the exact same value on multiple different machines at different times is quite small, so I don't see a need. I don't mind adding the userID but don't think it adds much to it.